### DISCUSSION TITLE
Supporting IGV

### DISCUSSION THREAD

**Discussion by mboisson** [ComputeCanada/magic_castle#102] [Ideas]
Integrative Genomic Viewer is a tool that can be used to visualize BAM files. This can be useful for bioinformaticians, to visualize large files rather than having to download them on their computer. 

http://software.broadinstitute.org/software/igv/

**Comment by zhibinlu** [comment]
In several CBW workshops, students align sequencing data to a reference genome and generate pretty big bam files. They used IGV (http://software.broadinstitute.org/software/igv/) to view the alignment. The bam file can be loaded from either local disk or an URL (http://software.broadinstitute.org/software/igv/LoadData). IGV will not load the whole file but only the portion it needs, so if a URL is provided, it only downloads a small portion of the file. This can be done with a web server or ftp server. When we used AWS, we install a web server (this is not secure of course), and students can access their files directly via URLs. I do not think this is possible with jupyterhub. We asked the students to download the bam files to their own machines and then load into IGV. Students complained it cost them too long to download the files especially when 30 students downloaded files from the login node at the same time.

**Comment by mboisson** [comment]
I suppose the access to files through URLs in IGV on the user's desktop is fundamentally a no-go, because this requires *unauthenticated* access to files by URL. Running IGV-web locally on the compute node (and proxy it through JH), or running the Java application through webvnc seems like a more viable route.

**Comment by zhibinlu** [comment]
I totally agree if you say this is a security issue. We installed web server on AWS because we gave each user a separate VM and the data set was public accessible. In fact, we mentioned to the students, a URL to the file under their home would not be possible in the real world. It can be a solution if we can launch IGV within jupyterhub like RStudio, but we want students to be able to use IGV after the workshop. So maybe downloading the bam files to their local machines is the best solution for now.

**Comment by poquirion** [comment]
The latest globus release lets us create urls for files on the system. You need some payed version for that, but CC/CQ does have that version. Globus is already part of Magic Castle, we would need to find a way (automatic or semi automatic) to register the Globus node with the official globus licence and Voilà, free web links on the file system for workshops.

This setup would also have the virtue of being reproducible on CC systems once Globus is updated there too.

**Comment by mboisson** [comment]
I know it's been a while, but is this still relevant ?
